docs/content/3.cookbook/2.pass-custom-headers-and-cookies-to-requests.md

```diff
@@ -0,0 +1,266 @@
+---
+title: Pass Custom Headers and Cookies to Requests
+layout: default
+scope: sdk, middleware, integrations
+---
+
+# Pass Custom Headers and Cookies to Requests
+
+In this tutorial, we'll demonstrate how to set up an interceptor to pass headers when making requests using the axios client inside the Vue Storefront SDK. This is useful when you need to append custom headers or cookies to your API calls.
+
+After passing the headers to the SDK client, we'll also show you how to pass them further from the Server Middleware to external services (e.g. SAP OCC API).
+
+## Notes
+::info
+This guide uses the integrations boilerplate with the nuxt option. The same basic principles apply to other frameworks, but the implementation may vary.
+::
+
+All SDK modules export a client instance of axios. This is the axios instance that is used to make requests to the Server Middleware. You can access this instance by importing the `client` from the SDK.
+
+## Steps
+
+step 1.
+###  **Import Required Modules**
+
+```javascript
+import { initSDK, buildModule } from '@vue-storefront/sdk';
+import { client, boilerplateModule, BoilerplateModuleType } from '../../packages/sdk/src';
```

Comment:
I don't understand, where is the client coming from, is it exposed by every SDK module? Is it coordinated with the unified team, do they expose it as well the same way?

Reply:
This import path should be less ambigious. But long story short: the client is coming from an SDK module (e.g. sapcc-sdk), every module exports the client. Not sure this is going to be the case in a few weeks/months (considering the ongoing discussion initiated by @bartoszherba) but, for now, that's the state of things.

Reply:
I really thing this should be agreed upon with unified. Consider how confused a customer on unified storefront would be after seeing these docs.

Reply:
This guide sat on the back burner for a while. Given the change in landscape since it's inception I agree there should be a consensus and standard way of doing this. RE: comments about ootb, I think it would be sensible for for the Unified team to implement the boilerplate to allow passing custom headers and allow the end dev to use a one-liner, if possible, to add headers to requests. The guide for an ootb solution would be a separate guide though. 

we still have two scenarios:

1.  Projects using Unified: for those we can have an ootb solution and a separate guide
2. Non-Unified:  this sort of guide will still be necessary.